tests/conftest.py

```diff
@@ -14,29 +14,40 @@ def auth_client(django_user_model) -> Client:
     return client
 
 
-@pytest.fixture
-def auth_process_client(django_user_model) -> Client:
-    """Return a authenticated client with modify process privilege."""
-    user = django_user_model.objects.create(username="process_user")
-    permission = Permission.objects.get(codename="can_modify_processes")
+def _privileged_user_client(django_user_model, username, permission_name):
+    user = django_user_model.objects.create(username=username)
+    permission = Permission.objects.get(codename=permission_name)
     user.user_permissions.add(permission)
     client = Client()
     client.force_login(user)
     return client
 
 
+@pytest.fixture
+def auth_process_client(django_user_model) -> Client:
+    """Return a authenticated client with modify process privilege."""
+    return _privileged_user_client(
+        django_user_model, "process_user", "can_modify_processes"
+    )
+
+
 @pytest.fixture
 def auth_logs_client(django_user_model) -> Client:
     """Return a authenticated client with view logs privilege."""
-    user = django_user_model.objects.create(username="logs_user")
-    permission = Permission.objects.get(codename="can_view_process_logs")
-    user.user_permissions.add(permission)
-    client = Client()
-    client.force_login(user)
-    return client
+    return _privileged_user_client(
+        django_user_model, "logs_user", "can_view_process_logs"
+    )
 
 
 @pytest.fixture
 def dummy_session_data() -> dict[str, str | int]:
     """A dictionary of dummy data to populate a dummy session."""
     return dict(session_name="sess_name", n_processes=1, sleep=5, n_sleeps=4)
+
+
+@pytest.fixture(autouse=True)
+def grpc_mock(mocker):
+    """Mock out the method that generates gRPC calls to external interfaces."""
+    yield mocker.patch(
+        "process_manager.process_manager_interface.ProcessManagerDriver.send_command_aio"
```

Comment:
Mocking specifically the part that generates the gRPC calls like this is good. We should now be able to add some tests that check the views are calling the correct methods of the ProcessManagerDriver

Reply:
Not sure where/how this is used.

Reply:
This is used automatically before every test is run - see the `fixture(autouse=True)` so we don't need to worry about mocking the calls to `gRPC` in every test that might use them.